### PR TITLE
Add Hardhat as a known node

### PR DESCRIPTION
**Pull request by fvictorio**
Fixes https://github.com/OpenZeppelin/openzeppelin-test-helpers/issues/137

This PR adds Hardhat Network as a known node.

I didn't add a version check for it because Hardhat will start at 2.0.0, so a semver check doesn't make sense.

I also didn't add any abstraction because there are only two nodes (I would wait for a third one before abstracting this, but YMMV).

cc @alcuadrado

### CODE REVIEW

File: src/expectRevert.js [hunk 1]
```diff
@@ -33,23 +33,24 @@ async function checkRevertReasonSupport (provider) {
...
       );
     };
 
-    if (ganacheVersion === null) {
+    if (ganacheVersion === null && hardhatVersion === null) {
       warn(`\
 Assertions may yield false negatives!
 
-Revert reason checks are only known to work on Ganache >=2.2.0, and the current node is ${nodeInfo}.
+Revert reason checks are only known to work on Ganache >=2.2.0 and Hardhat >= 2.0.0, and the current node is ${nodeInfo}.
```
Comment (alcuadrado) [inline src/expectRevert.js:48]: I think we don't need a version number for Hardhat, as all of them support it

Comment (fvictorio) [reply src/expectRevert.js:48]: Yeah, I initially added it for consistency but it doesn't really make sense. I will remove it.

**Comment by frangio** [comment]
@fvictorio Do you think you could add an integration test for this?

You would need to create a new project under `test-integration`, and copy the [`run.sh`](https://github.com/OpenZeppelin/openzeppelin-test-helpers/blob/89eb931c33f039f52a222f884555e652f6938e23/test-integration/ganache-core-2.1.x/run.sh) file that is used by the other integration tests. It just installs the version that is being tested and then runs `npm test`. (We shouldn't need to copy this bash file since it's the same always but it's set up this way now, I'll make a mental note to improve that later.) Let me know if I need to explain something else.

As a side note, has the Hardhat rename been announced already? Trying to understand whether it would make sense to publish this as soon as it's merged.

**Comment by fvictorio** [comment]
> Do you think you could add an integration test for this?

Sure, will do!



> As a side note, has the Hardhat rename been announced already?

No, it's not been announced yet.

**Comment by fvictorio** [comment]
Oh, I think adding this integration test will make more sense once all the packages are published, at least in RC version. I will add it as soon as we do that.

**Comment by fvictorio** [comment]
@frangio I added one integration test for Hardhat.

There is another test that is failing, at least for me locally, but it doesn't seem related to this PR (it fails for me in the master branch too).

**Review by frangio — APPROVED** [review]
Thank you @fvictorio!

**Comment by frangio** [comment]
Just released this in the latest version.

**Pull request merged**